src/test/versionbits_tests.cpp

```diff
@@ -443,8 +417,18 @@ BOOST_AUTO_TEST_CASE(versionbits_computeblockversion)
     // ACTIVE and FAILED states in roughly the way we expect
     for (const auto& chain_name : {CBaseChainParams::MAIN, CBaseChainParams::TESTNET, CBaseChainParams::SIGNET, CBaseChainParams::REGTEST}) {
         const auto chainParams = CreateChainParams(*m_node.args, chain_name);
+        std::set<decltype(Consensus::BIP9Deployment::bit)> chain_bits;
+        std::set<uint32_t> chain_masks;
         for (int i = 0; i < (int)Consensus::MAX_VERSION_BITS_DEPLOYMENTS; ++i) {
-            check_computeblockversion(chainParams->GetConsensus(), static_cast<Consensus::DeploymentPos>(i));
+            const auto dep = static_cast<Consensus::DeploymentPos>(i);
+            // Check that no bits are re-used (within the same chain). This is
+            // disallowed because the transition to FAILED (on timeout) does
+            // not take precedence over STARTED/LOCKED_IN. So all softforks on
+            // the same bit might overlap, even when non-overlapping start-end
+            // times are picked.
+            BOOST_CHECK(chain_bits.insert(chainParams->GetConsensus().vDeployments[dep].bit).second);
```

Comment:
Do we have any guidance on whether we should avoid side-effects in `BOOST_CHECK()` expressions?  This can obviously be rewritten in a way that inserts and then checks the return value on the following line, but I'm not sure if that's preferred

Reply:
Removed this line because I think it is redundant